chef/cookbooks/ses/providers/config.rb

```diff
@@ -0,0 +1,88 @@
+action :create do
+  create_configs(new_resource.name)
+end
+
+def supported_services
+  ["glance", "cinder", "nova"]
+end
+
+# TODO: use 'ceph' group for ownership of keyring files and add relevant service users
+# to this group instead of using per-service groups. This should cover case where multiple
+# services use the same user/keyring.
+
+# TODO: do we need some service restarting/reloading logic here? maybe some
+# (smart) detection if configs were changed? or is this handled by template somehow?
+
+def write_keyring_file(client_name, keyring_path, key, group_name)
+  Chef::Log.info("SES create #{client_name} keyring #{keyring_path}")
+
+  template keyring_path do
+    cookbook "ses"
+    source "client.keyring.erb"
+    owner "root"
+    group group_name
+    mode "0640"
+    variables(client_name: client_name,
+              keyring_value: key)
+  end
+end
+
+def create_configs(ses_service)
+  # This function creates the /etc/ceph/ceph.conf
+  # and the keyring files needed by the services
+  # ses_service is the name of the service using ceph
+  # which should be nova, cinder, glance
+  Chef::Log.info("SES: create_configs for service #{ses_service}")
+
+  ses_config = SesHelper.ses_settings
+  Chef::Log.debug("SES config = #{ses_config}")
+
+  # No SES config found? Do nothing.
+  return if ses_config.nil? || ses_config.empty?
+
+  Chef::Log.info("External SES configuration found.")
+
+  # supported services on current node
+  local_services = supported_services.select { |service| node.key? service }
+  # ses configs matching local services
+  local_ses_services = ses_config.select { |key| local_services.include?(key) }
+
+  # First create a unique clients list, so we don't have dupes in
+  # the ceph.conf file. This could happen if multiple services use one user.
+  ses_clients = {}
+  local_ses_services.each do |service, service_config|
+    user = service_config["rbd_store_user"]
+    ses_clients[user] = {
+      keyring: SesHelper.keyring_path(user),
+      key: service_config["key"]
+    }
+  end
+
+  # Make sure parent directory exists (only last level is checked here)
+  directory ::File.dirname(SesHelper.ceph_conf_path) do
+    owner "root"
+    group "root"
+    mode "0755"
+    action :create
+  end
+
+  Chef::Log.info("SES create #{SesHelper.ceph_conf_path}")
+  template SesHelper.ceph_conf_path do
+    cookbook "ses"
+    source "ceph.conf.erb"
+    owner "root"
+    group "root"
```

Comment:
This should be owned by the ceph group too.

Reply:
done